modules/calib3d/src/calibinit.cpp

```diff
@@ -504,10 +506,11 @@ bool findChessboardCorners(InputArray image_, Size pattern_size,
     int prev_sqr_size = 0;
 
     Mat thresh_img_new = img.clone();
-    icvBinarizationHistogramBased(thresh_img_new); // process image in-place
+    if(!is_plain)
+        icvBinarizationHistogramBased(thresh_img_new); // process image in-place
     SHOW("New binarization", thresh_img_new);
 
-    if (flags & CALIB_CB_FAST_CHECK)
+    if (flags & CALIB_CB_FAST_CHECK && !(flags & CALIB_CB_PLAIN))
```

Comment:
```suggestion
    if (flags & CALIB_CB_FAST_CHECK && !is_plain)
```

Reply:
Hi mshabunin I'm not so sure about the idea of having both CALIB_CB_PLAIN and CALIB_CB_FAST_CHECK active at the same time.
The basic idea with CALIB_CB_PLAIN is to have a function that in fact does "nothing" in the sense that all image processing (binarization) is done externally